src/software/ai/navigator/path_planner/hrvo/simulated_hrvo_test.py

```diff
@@ -0,0 +1,462 @@
+        (
+            [Point(x_meters=2.8, y_meters=-1)],
+            [Point(x_meters=2.8, y_meters=1)],
+            [],
+            [
+                Point(x_meters=float(x_meters) / 10, y_meters=0)
+                for x_meters in range(20, 32, 2)
+            ],
+            [],
+            25,
+        ),
+        # robot moving in a local minima (enemy robots in a curve around it)
+        (
+            [Point(x_meters=0.7, y_meters=0)],
+            [Point(x_meters=2.7, y_meters=1)],
+            [],
+            [
+                Point(x_meters=1, y_meters=0),
+                Point(x_meters=1, y_meters=0.3),
+                Point(x_meters=1, y_meters=0.6),
+                Point(x_meters=0.7, y_meters=0.6),
+                Point(x_meters=1, y_meters=-0.3),
+                Point(x_meters=1, y_meters=-0.6),
+                Point(x_meters=0.7, y_meters=-0.6),
+            ],
+            [],
+            25,
+        ),
+        # robot moving in a local minima (enemy robots in a curve around it) with an opening in the middle
+        (
+            [Point(x_meters=0.7, y_meters=0)],
+            [Point(x_meters=2.7, y_meters=1)],
+            [],
+            [
+                Point(x_meters=1.5, y_meters=0),
+                Point(x_meters=1, y_meters=0.3),
+                Point(x_meters=1, y_meters=0.6),
+                Point(x_meters=0.7, y_meters=0.6),
+                Point(x_meters=1, y_meters=-0.3),
+                Point(x_meters=1, y_meters=-0.6),
+                Point(x_meters=0.7, y_meters=-0.6),
+            ],
+            [],
+            10,
+        ),
+        # robot moving in a zig zag path around enemy robots
+        get_zig_zag_params(-2, 0.3, 3),
+        # friendly robots in a circle moving along each diameter
+        (
+            get_robot_circle_pos(1.5, 8, True),
+            get_robot_circle_pos(1.5, 8, False),
+            [],
+            [],
+            [],
+            15,
+        ),
+        # friendly robots in a circle moving along each diameter
+        # while turning from 0 to 180 degrees
+        (
+            get_robot_circle_pos(1.5, 8, True),
+            get_robot_circle_pos(1.5, 8, False),
+            [Angle(radians=math.pi) for i in range(8)],
+            [],
+            [],
+            20,
+        ),
+        # half enemy half friendly robots in a circle moving along each diameter
+        (
+            [
+                pos
+                for index, pos in enumerate(get_robot_circle_pos(1.5, 8, True))
+                if index % 2 == 0
+            ],
+            [
+                pos
+                for index, pos in enumerate(get_robot_circle_pos(1.5, 8, False))
+                if index % 2 == 0
+            ],
+            [],
+            # both the start and end positions are the same values
+            # but for some reason it works when actually running the test
+            # so just gonna leave this here
```

Comment:
It would be good to figure this out if it's not too complex just to clear any future confusions with the test setups